packages/sdk/src/pipeline.ts

```diff
@@ -0,0 +1,83 @@
+import { get, post, put, del, listen } from './request';
+import { tunaMirrorURI } from './utils';
+import { PipelineModel, PipelineInstallOption } from './interface';
+
+/**
+ * API for pipeline
+ */
+export class Pipeline {
+  route: string;
+  constructor(url: string) {
+    this.route = `${url}/pipeline`;
+  }
+
+  /**
+   * list all pipelines
+   */
+  async list(): Promise<PipelineModel[]> {
+    return (await get(`${this.route}/list`)).rows;
+  }
+
+  /**
+   * get pipeline info by pipeline id
+   * @param id pipeline id
+   */
+  async info(id: string): Promise<PipelineModel> {
+    return await get(`${this.route}/info/${id}`);
+  }
+
+  /**
+   * create a pipeline by pipeline config object
+   * @param config pipeline config
+   * @param opts name: pipeline name
+   */
+  async create(config: object, opts: any): Promise<PipelineModel> {
+    return await post(`${this.route}`, {
+      config,
+      name: opts.name,
+      isFile: false
+    });
+  }
+
+  /**
+   * update pipeline config
+   * @param id pipeline id
+   * @param config pipeline config
+   */
+  async update(id: string, config: object): Promise<PipelineModel> {
+    return await put(`${this.route}/${id}`, {
+      config,
+      isFile: false
+    });
+  }
+
+  /**
+   * remove pipeline by id, if the id is undefined or 'all', remove all
+   * @param id pipline id or 'all'
+   */
+  async remove(id?: any): Promise<number> {
+    if (typeof id === 'string' && id !== 'all') {
+      return await del(`${this.route}/${id}`);
+    } else {
+      return await del(this.route);
+    }
+  }
+
+  /**
+   * install plugins defined in pipeline
+   * @param id pipeline id
+   * @param opt tuna: is using tuna mirror
+   */
+  async install(id: string, opt?: PipelineInstallOption): Promise<void> {
```

Comment:
```suggestion
  async installPlugins(id: string, opt?: PipelineInstallOption): Promise<void> {
```

Reply:
And should we provide the step function when a plugin is installed?

Reply:
Could we comment this as TODO?